### PR TITLE
sioyek: add module

### PR DESCRIPTION
**Pull request by podocarp**
### Description
Adds the module for the pdf reader sioyek.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/sioyek.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        '';
+        type = types.attrsOf types.str;
+        default = { };
+        example = literalExpression ''
+          {
+          "move_up" = "k";
+          "move_down" = "j";
+          "move_left" = "h";
+          "move_right" = "l";
+          }
```
Comment (berbiche) [inline modules/programs/sioyek.nix:36]:
```suggestion
          {
            "move_up" = "k";
            "move_down" = "j";
            "move_left" = "h";
            "move_right" = "l";
          }
```
nit: indentation

File: modules/programs/sioyek.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          <filename>$XDG_CONFIG_HOME/sioyek/prefs_user.config</filename>.
+          See <link xlink:href="https://github.com/ahrm/sioyek/blob/main/pdf_viewer/prefs.config"/>.
+        '';
+        type = types.attrsOf types.str;
+        default = { };
+        example = literalExpression ''
+          {
+          "background_color" = "1.0 1.0 1.0";
+          "text_highlight_color" = "1.0 0.0 0.0";
+          }
```
Comment (berbiche) [inline modules/programs/sioyek.nix:52]:
```suggestion
          {
            "background_color" = "1.0 1.0 1.0";
            "text_highlight_color" = "1.0 0.0 0.0";
          }
```

File: modules/programs/sioyek.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    { home.packages = [ cfg.package ]; }
+    (mkIf (cfg.config != { }) {
+      xdg.configFile."sioyek/prefs_user.config".text = renderAttrs cfg.config;
+    })
+    (mkIf (cfg.bindings != { }) {
+      xdg.configFile."sioyek/keys_user.config".text = renderAttrs cfg.bindings;
+    })
+  ]);
+
+  meta.maintainers = [ maintainers.podocarp ];
```
Comment (berbiche) [inline modules/programs/sioyek.nix:69]:
```suggestion
  meta.maintainers = [ hm.maintainers.podocarp ];
```

Home Manager's maintainer list is in `lib.hm.maintainers`.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!
I've highlighted some small nits that need to be fixed, otherwise everything is fine.

File: modules/programs/sioyek.nix (hunk 1 above)
Comment (podocarp) [reply modules/programs/sioyek.nix:36]: Ok, thanks. I copied the example from the mpv module, and noticed that they did not have indentation, so perhaps someone should make a small edit there as well.

**Comment by podocarp** [comment]
Thanks for the comments! Commited the changes.
Just a question, what is the difference between `maintainers` and `hm.maintainers`?
Just using `maintainers` seems to compile fine. Is that pointing to the nixpkgs maintainers instead?

**Review by berbiche — APPROVED** [review]
LGTM!

**Pull request merged**